src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java

```diff
@@ -1,39 +1,40 @@
 package edu.kis.vh.nursery;
 
+import edu.kis.vh.nursery.int_data_structure.IntArrayStack;
+import edu.kis.vh.nursery.int_data_structure.IntDataStructure;
+
 public class DefaultCountingOutRhymer {
-    private static final int minusOne = -1;
-    private static final int eleven = 11;
-    private static final int twelve = 12;
-    private final int[] numbers = new int[twelve];
-    private int total = minusOne;
+    IntDataStructure intArrayStack = null;
+
+    public DefaultCountingOutRhymer() {
+        intArrayStack = new IntArrayStack();
+    }
+
+    public DefaultCountingOutRhymer(IntArrayStack intArrayStack) {
```

Comment:
The constructor should take StructInterface as an argument. It would be the most consistent and flexible approach.

The goal of the lab was to make using IntLinkedList and IntArrayStack interchangeable.